src/sentry/models/release.py

```diff
@@ -257,14 +257,17 @@ def set_refs(self, refs, user, fetch=False):
             projects__in=self.projects.all(),
         ).exclude(version=self.version).order_by('-date_added').first()
 
+        names = {r['repository'] for r in refs}
+        repos = list(Repository.objects.filter(
+            organization_id=self.organization_id,
+            name__in=names,
+        ))
+        repos_by_name = {r.name: r for r in repos}
+        if len(repos_by_name) != len(names):
+            raise Repository.DoesNotExist
```

Comment:
I'd raise a different exception, then we can provide information about which repos are invalid to use in the response we return.